internal/commands/http_test.go

```diff
@@ -0,0 +1,333 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/buildkite/buildkite-mcp-server/pkg/server"
+	"github.com/buildkite/go-buildkite/v4"
+	mcpserver "github.com/mark3labs/mcp-go/server"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+)
+
+// MockHTTPServer interface that both SSE and Streamable servers implement
+type MockHTTPServer interface {
+	Start(addr string) error
+}
+
+// MockSSEServer simulates the SSE server for testing
+type MockSSEServer struct {
+	StartFunc  func(addr string) error
+	StartCalls []string // Track calls made to Start
+}
+
+func (m *MockSSEServer) Start(addr string) error {
+	m.StartCalls = append(m.StartCalls, addr)
+	if m.StartFunc != nil {
+		return m.StartFunc(addr)
+	}
+	return nil
+}
+
+// MockStreamableServer simulates the streamable HTTP server for testing
+type MockStreamableServer struct {
+	StartFunc  func(addr string) error
+	StartCalls []string // Track calls made to Start
+}
+
+func (m *MockStreamableServer) Start(addr string) error {
+	m.StartCalls = append(m.StartCalls, addr)
+	if m.StartFunc != nil {
+		return m.StartFunc(addr)
+	}
+	return nil
+}
+
+// ServerFactory interface for creating servers (used for dependency injection in tests)
+type ServerFactory interface {
+	NewSSEServer(server *mcpserver.MCPServer) MockHTTPServer
+	NewStreamableHTTPServer(server *mcpserver.MCPServer) MockHTTPServer
+}
+
+// MockServerFactory implements ServerFactory for testing
+type MockServerFactory struct {
+	SSEServer        *MockSSEServer
+	StreamableServer *MockStreamableServer
+}
+
+func (f *MockServerFactory) NewSSEServer(server *mcpserver.MCPServer) MockHTTPServer {
+	return f.SSEServer
+}
+
+func (f *MockServerFactory) NewStreamableHTTPServer(server *mcpserver.MCPServer) MockHTTPServer {
+	return f.StreamableServer
+}
+
+// TestableHTTPCmd extends HTTPCmd to allow dependency injection for testing
+type TestableHTTPCmd struct {
+	HTTPCmd
+	ServerFactory ServerFactory
+}
+
+func (c *TestableHTTPCmd) Run(ctx context.Context, globals *Globals) error {
```

Comment:
The tests below appear to just be testing this struct that isn't actually used in the application itself. I think I'll blow them all away and loop back in a subsequent PR and write some integration tests that are a bit further out in the system, and exercise the real command with a real client.